alacritty/src/display/mod.rs

```diff
@@ -193,11 +194,33 @@ pub struct Display {
     /// Unprocessed display updates.
     pub pending_update: DisplayUpdate,
 
+    /// Unprocessed IME input.
+    pub ime_input: Option<IMEInput>,
+
     renderer: QuadRenderer,
     glyph_cache: GlyphCache,
     meter: Meter,
 }
 
+/// A state of IME input into the window.
+#[derive(Debug, Default)]
+pub struct IMEInput {
+    /// The text being preedit.
+    preedit: String,
+
+    /// Byte offset for cursor start in text.
+    cursor_start: Option<usize>,
+
+    /// Byte offset for cursor end in text.
+    _cursor_end: Option<usize>,
```

Comment:
This is basically to have a highlight in preedit string while you're at it. We could probably do that by drawing hollow block cursor from cursor_start up to cursor_end, what do you think? For now I don't handle end.

Reply:
What is start/end here? In which scenario can the cursor be more than a fullwidth character (I'm not familiar with IME)?

Reply:
The Cursor covers the whole sentence, often one writes whole sentences:
E.G:
Start: 0, end: 0 = |您好
Start: 0, end: 6* = |您好 (or the whole text is marked or something, I haven't seen it yet)
Start: 6, end: 6* = 您好|

*(these are usually 3 bytes each)

Reply:
Yeah, it could be treated as selection from start to end in preedit, I also haven't seen something like this, all IMEs I've used just report the same start and end.